fix `_owner` nullability in `routers.dart`

TahaTesser opened the pull request:
fixes https://github.com/flutter/flutter/issues/84769

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

TahaTesser committed:
fix _owner nullability in the remove()

Piinks reviewed (commented):
Hi @TahaTesser thanks for the fix! This should have a test. Also, can you rebase your branch? That should resolve the failing checks from an infra issue.

Piinks commented:
(triage) Would you like to continue with this change @TahaTesser ?

TahaTesser commented:
@Piinks 
Thanks for the review, I would love to make the changes soon)

TahaTesser committed:
Merge branch 'master' of github.com:flutter/flutter into fix_owner_nullability

TahaTesser committed:
Merge branch 'master' of github.com:flutter/flutter into fix_owner_nullability

TahaTesser committed:
\Merge branch 'master' of https://github.com/flutter/flutter into fix_owner_nullability

TahaTesser committed:
dispose localhistory test

TahaTesser committed:
Merge branch 'fix_owner_nullability' of github.com:TahaTesser/flutter into fix_owner_nullability

TahaTesser committed:
fix new line

TahaTesser commented:
@Piinks 
Apologies for the delay, I had other commitments,
I have added a test and widget class for the test and rebased the branch
If the `_owner` is not nullable in `remove()` - test fails (as expected) and `_owner` is nullable  - test passes (as expected) as the change here is to make `_owner` nullable.

Thank you so much for the review

TahaTesser committed:
remove whitespaces

skia-gold commented:
Gold has detected about 1 new digest(s) on patchset 8.
View them at https://flutter-gold.skia.org/cl/github/84840

goderbauer commented on packages/flutter/test/widgets/routes_test.dart:
```diff
@@ -1700,8 +1700,8 @@ void main() {
       // Waits for modal route to update its internal state;
-      await tester.pump();
 
+      await tester.pump();
```
nit: revert this change here? This line belongs to the comment above and there shouldn't be a blank line in between.

goderbauer commented on packages/flutter/test/widgets/routes_test.dart:
```diff
@@ -1712,6 +1712,25 @@ void main() {
       await tester.pump();
       expect(tester.takeException(), null);
     });
+    testWidgets('child with no local history can be disposed', (WidgetTester tester) async {
```
nit: add a blank line before this one.

goderbauer commented on packages/flutter/test/widgets/routes_test.dart:
```diff
@@ -1712,6 +1712,25 @@ void main() {
+      final WidgetWithNoLocalHistoryState state = tester.state(find.byType(WidgetWithNoLocalHistory));
+      state.addLocalHistory();
+      // Waits for modal route to update its internal state;
+
```
nit: remove this blank line or place it before the comment

goderbauer reviewed (approved):
LGTM after nits are resolved.

TahaTesser committed:
Merge branch 'master' of github.com:flutter/flutter into fix_owner_nullability

TahaTesser committed:
nit fixes

TahaTesser commented:
@goderbauer 
Thanks so much for the review,, made the changes https://github.com/flutter/flutter/pull/84840/commits/444045ba07091666a4da28887065df8ac455bcb7

goderbauer reviewed (approved):
LGTM
